docker-images/kafka-connect/scripts/kafka_connect_config_generator.sh

```diff
@@ -42,9 +42,19 @@ if [ -n "$KAFKA_CONNECT_SASL_USERNAME" ] && [ -n "$KAFKA_CONNECT_SASL_PASSWORD_F
     else
         SECURITY_PROTOCOL="SASL_PLAINTEXT"
     fi
-
+    
     PASSWORD=$(cat /opt/kafka/connect-password/$KAFKA_CONNECT_SASL_PASSWORD_FILE)
-    SASL_MECHANISM="SCRAM-SHA-512"
+
+    if [ "x$KAFKA_CONNECT_SASL_MECHANISM" = "xplain" ]; then
+        SASL_MECHANISM="PLAIN"
+
+        # SSL encryption should always be used if SASL mechanism is PLAIN
+        SECURITY_PROTOCOL="SASL_SSL"
```

Comment:
I don't think this should be here. This is already set on lines 41 or 43.

Reply:
lines 41 or 43 $SECURITY_PROTOCOL (SASL_SSL / SASL_PLAINTEXT) is relying on lines 5 - 6
```
if [ -n "$KAFKA_CONNECT_TRUSTED_CERTS" ]; then
    SECURITY_PROTOCOL="SSL"
```
In that case if we do not feed KAFKA_CONNECT_TRUSTED_CERTS, the protocol would fallback to "SASL_PLAINTEXT".

However in our scenario we do need "SASL_SSL" protocol. The $KAFKA_CONNECT_TRUSTED_CERTS check doesn't work for us.

Actually I added the comment of "SSL encryption should always be used if SASL mechanism is PLAIN" which is based here https://docs.confluent.io/current/kafka/authentication_sasl/authentication_sasl_plain.html

Please advise and correct me if I missed something here ...

Reply:
> Actually I added the comment of "SSL encryption should always be used if SASL mechanism is PLAIN" which is based here https://docs.confluent.io/current/kafka/authentication_sasl/authentication_sasl_plain.html

Well, this is valid point. But I don't think we cannot let the user configure SASL PLAIN without TLS in the custom resource and then just force the SASL_SSL on it. such configuration would definitely not work because it would be missing the certificates. It would also do something different from what the user asked for. So I think there are two ways how to solve this properly:

1) Detect it properly inside the `KafkaConnectCluster` and throw an `InvalidResourceException` exception.
2) Just add a warning to the docs that this is dangerous but let the users decide what is best for him.

I would probably prefer 2, but can live with 1 as well.

Reply:
After discussion, removed line 51-52 hardcoded SASL_SSL for plain type, 
commit 0eb3875 
( after tls enhancement https://github.com/rabbitzzy/strimzi-kafka-operator/pull/1 merged)